pkg/util/deployment/deployment.go

```diff
@@ -409,3 +420,52 @@ func ResolveFenceposts(maxSurge, maxUnavailable *intstrutil.IntOrString, desired
 
 	return surge, unavailable, nil
 }
+
+// used for unit testing
+var nowFn = func() time.Time { return time.Now() }
+
+// TimedOut returns if the provided deployment is progressing or timed out.
```

Comment:
This is ambiguous; suggest `... returns true if ... timed out, false otherwise.`